napari/utils/transforms/transforms.py

```diff
@@ -135,11 +157,15 @@ def inverse(self) -> 'TransformChain':
         return TransformChain([tf.inverse for tf in self[::-1]])
 
     @property
-    def _is_diagonal(self):
+    def _is_diagonal(self) -> bool:
         if all(getattr(tf, '_is_diagonal', False) for tf in self):
             return True
         return getattr(self.simplified, '_is_diagonal', False)
```

Comment:
```suggestion
        return all(tf._is_diagonal for tf in self)
```

As `_is_diagonal` is a property defined on the base class that has a default implementation, can we simplify this? Or maybe `return self.simplified._is_diagonal`?